viewing_party/party.py

```diff
@@ -1,23 +1,138 @@
+from statistics import mean, mode
+# -----------------------------------------
 # ------------- WAVE 1 --------------------
-
+# -----------------------------------------
+# If valid, creates a movie with title, genre, and rating as values of a dict
 def create_movie(title, genre, rating):
-    pass
+    if title == None or genre == None or rating == None:
+        return None
+    else:
+        movie = {"title" : title, "genre" : genre, "rating" : rating}
+    return movie
+
+# Adds movie dictionary to watched
+def add_to_watched(user_data, movie):
+    user_data = {}
+    user_data["watched"] = [movie]
+    return user_data
+
+# Adds movie dictionary to watchlist
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+# If the title is in a movie in the user's watchlist: remove that movie from the watchlist, add that movie to watched
+def watch_movie(user_data, title):
+    watchlist = user_data["watchlist"]
+    for film in range(len(watchlist)):
+        if watchlist[film]["title"] == title:
+            user_data["watched"].append(watchlist[film])
+            del watchlist[film]
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+# Find and return avg of reviews in watched
+def get_watched_avg_rating(user_data):
+    reviews = []
+    for film in user_data["watched"]:
+        reviews.append(film["rating"])
+    if not reviews:
+        return 0
+    average = mean(reviews)
+    return average

```

Comment:
👍  A few thoughts:
- If you utilize `mean`  or any other imported package/module during an interview, be prepared to discuss how that function/method/class works under the hood. Anything that we use during an interview is fair game to being asked about!
- We can check for an empty watched list at the beginning of the function which would allow us to exit the function early. 
```python
def get_watched_avg_rating(user_data):
    if not user_data["watched"]:
         return 0 
         
    reviews = []
    for film in user_data["watched"]:
        reviews.append(film["rating"])
    average = mean(reviews)
    return average
```